src/main/webapp/app/exercises/programming/assess/manual-result/programming-assessment-manual-result-dialog.component.ts

```diff
@@ -137,7 +140,15 @@ export class ProgrammingAssessmentManualResultDialogComponent implements OnInit
                 tap(({ body: participation }) => {
                     this.participation = participation! as ProgrammingExerciseStudentParticipation;
                     this.result.participation = this.participation;
-                    this.isOpenForSubmission = this.participation.exercise.dueDate === null || this.participation.exercise.dueDate.isAfter(moment());
+
+                    if (!!this.exercise.exerciseGroup) {
+                        const exam = this.participation.exercise.exerciseGroup!.exam!;
```

Comment:
Not 100% sure, but I think unwrapping `exerciseGroup` should be implicit by if statement?
```suggestion
                        const exam = this.participation.exercise.exerciseGroup.exam!;
```

Reply:
I get a warning with "Object is possibly null". Therefore, I would leave it :)

Reply:
🆗